parts/k8s/kubernetesmastergenerateproxycertscript.sh

```diff
@@ -43,14 +52,19 @@ write_certs_to_disk_with_retry() {
         [ $? -eq 0  ] && break || sleep 5
     done
 }
-
+is_cluster_healthy(){
+    for i in 1 2 3 4 5 6 7 8 9 10 11 12; do
+        ETCDCTL_API=3 etcdctl ${ETCDCTL_PARAMS} endpoint health
+        [ $? -eq 0  ] && break || sleep 5
+    done
+}
 # block until all etcd is ready
-retrycmd_if_failure 100 5 10 etcdctl cluster-health
+is_cluster_healthy 
```

Comment:
why not use the already existing `retrycmd_if_failure` function?

Reply:
`retrycmd_if_failure 12 5 10 ETCDCTL_API=3 etcdctl ${ETCDCTL_PARAMS} endpoint health` would do the job

Reply:
because `retrycmd_if_failure` does not work with bash functions, only command line calls. I tried :-)

Reply:
why do you need a function? `retrycmd_if_failure 12 5 10 ETCDCTL_API=3 etcdctl ${ETCDCTL_PARAMS} endpoint health` would work no?